### PR TITLE
Add documentation for #17573

### PR DESCRIPTION
**Pull request by maclover7**
Fixes some parts of #23148.

[ci skip]

### CODE REVIEW

File: actionpack/lib/action_dispatch/http/cache.rb [hunk 1]
```diff
@@ -80,6 +80,11 @@ def date=(utc_time)
           set_header DATE, utc_time.httpdate
         end
 
+        # By default, Action Dispatch sets all ETags to be weak.
+        # This is so that way, in case the page changes only
```
Comment (vipulnsward) [inline actionpack/lib/action_dispatch/http/cache.rb]: `..so that, in case..`.  
`way` is confusing here.

Comment (vipulnsward) [inline actionpack/lib/action_dispatch/http/cache.rb]: Actually better would be 
`This ensures that when the page change semantically, the whole page doesn't have to be regenerated.`
or just
`This way when the page change semantically, the whole page doesn't have to be regenerated.`

File: actionpack/lib/action_dispatch/http/cache.rb [hunk 2]
```diff
@@ -80,6 +80,11 @@ def date=(utc_time)
           set_header DATE, utc_time.httpdate
         end
 
+        # By default, Action Dispatch sets all ETags to be weak.
+        # This ensures that if/when the page change semantically,
```
Comment (vipulnsward) [inline actionpack/lib/action_dispatch/http/cache.rb]: `changes`

Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/http/cache.rb]: This is not a documentation for a method but a comment on how it works in the HTTP point of view. I think this is not the best place to put this kind of information.

**Pull request merged**